typings/index.d.ts

```diff
@@ -1,95 +1,95 @@
-// Type definitions for CodeceptJS 1.0.2
+// Type definitions for CodeceptJS
 // Project: https://github.com/codeception/codeceptjs/
-// Definitions by: Michael Bodnarchuk <http://github.com/DavertMik>, Drew Diamantoukos <https://github.com/KennyRules>
+/// <reference path="./types.d.ts" />
 
-declare class Locator {
-  or(locator): Locator;
-  find(locator): Locator;
-  withChild(locator): Locator;
-  find(locator): Locator;
-  at(position): Locator;
-  first(): Locator;
-  last(): Locator;
-  inside(locator): Locator;
-  before(locator): Locator;
-  after(locator): Locator;
-  withText(text): Locator;
-  withAttr(attr): Locator;
-  as(output): Locator;
-}
+import * as Protractor from "protractor";
+import index = require("../lib/index");
 
-declare module NodeJS {
-  interface Process {
-    profile: string;
-  }
+type ValueOf<T> = T[keyof T]
+type KeyValueTupleToObject<T extends [keyof any, any]> = {
+  [K in T[0]]: Extract<T, [K, any]>[1]
+}
+export type Translate<T, M extends Record<string, string>> = KeyValueTupleToObject<ValueOf<{
+  [K in keyof T]: [K extends keyof M ? M[K] : K, T[K]]
+}>>
 
-  interface Global {
-    codecept_actor: any;
-    codecept_dir: string;
-    codecept_helper: any;
-    output_dir: string;
+export type ActorStatic = {
+  say: (msg: any, color?: any) => Promise<any> | undefined,
+  retry: (opts: any) => any
+}
 
-    actor: any;
-    Helper: any;
-    pause: any;
-    within: any;
-    session: any;
-    DataTable: any;
-    locate: Locator,
-    inject: any,
-    secret: any,
-    by: any;
+type codeceptjs = {
+  codecept: typeof Codecept,
+  container: typeof Container,
+  config: typeof Config,
+  actor: any,
+  helper: typeof Helper,
+  pause: typeof pause,
+  within: typeof within,
+  dataTable: typeof DataTable,
+  locator: typeof Locator,
+  
+  // export typings from JS files, type any when allowJs is set to false
+  recorder: typeof index.recorder,
+  event: typeof index.event,
+  output: typeof index.output,
+  store: typeof index.store,
+};
 
-    // Used by Protractor helper
-    By: any;
-    ExpectedConditions: any;
-    element: any;
-    $: any;
-    $$: any;
-    browser: any;
-  }
-}
+declare global {
+  const codeceptjs: codeceptjs;
+  
+  const codecept_dir: string;
+  const codecept_helper: Helper;
+  const output_dir: string;
 
-declare interface Window {
-  codeceptjs: any;
-}
+  const locate: typeof Locator.build;
+  const inject: typeof Container.support;
```

Comment:
Autocomplete doesnt work when I import so: `const { I, PageObject} = inject()`

Reply:
Should be fixed